src/EditorFeatures/Core/LanguageServer/AbstractInProcLanguageClient.cs

```diff
@@ -87,6 +88,7 @@ internal abstract partial class AbstractInProcLanguageClient : ILanguageClient,
             LspWorkspaceRegistrationService lspWorkspaceRegistrationService,
             ILspLoggerFactory lspLoggerFactory,
             IThreadingContext threadingContext,
+            DefaultCapabilitiesProvider defaultCapabilitiesProvider,
```

Comment:
Consider adding a TODO here or on the field to get rid of this once we clean up this hack properly; the original implementation here was nicely abstracted in that the implementation just override GetCapabilities and the base type didn't know where it was coming from. This is now a but funky in that you pass down a provider to have a method called...which then might be unnecessary if GetCapabilities did something else and never used DefaultCapabilitiesProvider in the first place!

Which, thinking about it, this might reveal a better abstraction here: if GetCapabilities was made async, and DefaultCapabilitiesProvider.GetCapabilities was also made async, then it can just switch back and forth on the threads as appropriate without this hack.

Reply:
Yeah, you make a good point. I originally did this without adding the default provider to the base class, and just had this happen in the actual clients before the GetCapabilities call, but it was just the same code everywhere, so I brought it back down. Only the Xaml server doesn't actually use the default, and AFAIK thats not in use yet. Will try to change this up.

Reply:
Looked into this, and its pretty ugly. Have left a comment to remove this instead.